extras/paho_mqtt_c/MQTTClient.c

```diff
@@ -94,8 +94,13 @@ int  readPacket(MQTTClient* c, Timer* timer)
         goto exit;
     len = 1;
     /* 2. read the remaining length.  This is variable in itself */
-    decodePacket(c, &rem_len, left_ms(timer));
-    len += MQTTPacket_encode(c->readbuf + 1, rem_len); /* put the original remaining length back into the buffer */
+    len += decodePacket(c, &rem_len, left_ms(timer));
```

Comment:
Similar issue was also reported in the upstream Paho client library (https://github.com/eclipse/paho.mqtt.embedded-c/issues/40), but it doesn't seem to be a popular project or getting much attentions. Also, the MQTT client library included here was forked, therefore merging is also going to be a problem.